### PR TITLE
Initial Block Guides pages

### PR DESCRIPTION
**Pull request by apple502j**

### CODE REVIEW

File: .vitepress/sidebars/develop.ts [hunk 1]
```diff
@@ -22,6 +22,16 @@ export default [
         link: "https://github.com/FabricMC/fabric-loom"
       }
     ]
+  },
+    {
```
Comment (modmuss50) [inline .vitepress/sidebars/develop.ts]: Nit, formatting.

File: develop/blocks/basics.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+Note that the item for your block is registered separately. Some blocks (like fire) do not have an item form, but most do. To register the corresponding item, create a `BlockItem` instance and register it under `Registries.ITEM`.
+
+@[code lang=java transcludeWith=:::3](@/reference/latest/src/main/java/com/example/docs/block/FabricDocsReferenceBlocks.java)
+
+## Block models and textures
+A block has a model, which describes the shape and textures of the rendered block. (This is different from the shape used for checking collision.) A block can have multiple models depending on the block state; the model is picked according to the "blockstate file". Items, including block-placing items, have separate models.
+
+Let's give the border block a texture with white background and black borders, like this. Most block textures are 16-by-16 square. This is placed at `/textures/block/border.png`, inside the mod resource pack (`/assets/(mod id)/`).
+
+![](/reference/latest/src/main/resources/assets/fabric-docs-reference/textures/block/border.png)
```
Comment (modmuss50) [inline develop/blocks/basics.md]: This is quite small on the web page, can the size be increased?

File: develop/blocks/basics.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+
+This texture is then referenced by a model file named `border.json`, placed under `/models/block` directory inside the mod resource pack. Since the block has the same texture on all six sides, we use `cube_all` as the parent model.
+
+@[code lang=json](@/reference/latest/src/main/resources/assets/fabric-docs-reference/models/block/border.json)
```
Comment (modmuss50) [inline develop/blocks/basics.md:47]: This is good so dont change anything, but I do think we want to encorage the use of data generation, especially as its likely going to be a lot more important in future updates.

File: develop/blocks/basics.md [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+@[code lang=java transcludeWith=:::2](@/reference/latest/src/main/java/com/example/docs/block/FabricDocsReferenceBlocks.java)
+
+Note that the item for your block is registered separately. Some blocks (like fire) do not have an item form, but most do. To register the corresponding item, create a `BlockItem` instance and register it under `Registries.ITEM`.
+
+@[code lang=java transcludeWith=:::3](@/reference/latest/src/main/java/com/example/docs/block/FabricDocsReferenceBlocks.java)
+
+## Block models and textures
+A block has a model, which describes the shape and textures of the rendered block. (This is different from the shape used for checking collision.) A block can have multiple models depending on the block state; the model is picked according to the "blockstate file". Items, including block-placing items, have separate models.
+
+Let's give the border block a texture with white background and black borders, like this. Most block textures are 16-by-16 square. This is placed at `/textures/block/border.png`, inside the mod resource pack (`/assets/(mod id)/`).
```
Comment (modmuss50) [inline develop/blocks/basics.md]: Maybe mention where the resources are? eg. `src/client/resources` ?

File: develop/blocks/basics.md (hunk 2 above)
Comment (modmuss50) [inline develop/blocks/basics.md]: Assets should be in src/client not src/main

File: develop/blocks/basics.md [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+
+@[code lang=java transcludeWith=:::4](@/reference/latest/src/main/java/com/example/docs/block/MachinePrototypeBlock.java)
+
+Here, we use another, very important method: `BlockState#get`, which returns the value for the property.
+
+::: tip
+Your code editor or compiler might output a "deprecation warning". These can be ignored - in reality, nothing is deprecated here.
+
+Why are they marked as deprecated, then? Here, it is how Mojang marks methods that are "override-only". Those methods should not be called by modded code (except in `super` call).
+:::
```
Comment (modmuss50) [inline develop/blocks/basics.md:93]: Note: It should be possible to remove these in 1.20.5 :)

**Review by modmuss50 — COMMENTED** [review]
This is a great start, there are checkstyle warnings in the code that need fixing and some minor comments.

File: develop/blocks/basics.md [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+title: Creating Blocks
+description: A guide for creating simple blocks.
+authors:
+  - apple502j
+---
+
+# Creating Blocks
+One of the most common things to do in your mod is adding custom blocks.
+
+A block, at its simplest form, has no functionality and only one "block state". A block state is a combination of properties, or variations, for the same block - things like rotations, crop age, or redstone activation state. For example, chains have 6 block states; 3 for the `axis` property, times 2 for the `waterlogged` property. A block can have only finite number of pre-defined block states. A block that needs more storage and functionality must have an associated "block entity". A block entity can store infinite, customizable data and perform logics every tick.
```
Comment (IMB11) [inline develop/blocks/basics.md]: See my overall review comment, this is a prime example of a bulky paragraph, you should really split this up - see what I mean through this suggestion, it's easier to read and less repetitive, you keep saying "A block" quite a bit haha :)

```suggestion
Blocks in their most basic form are static objects with a single appearance. To gain variations (like rotation, growth stage, or activation state), a block uses "block states". 

A block can only have a limited number of these predefined states. If a block needs more complex data storage or the ability to perform actions, it requires an associated "block entity", which offers customizable storage and the capacity to run logic each tick.
```

**Review by IMB11 — CHANGES_REQUESTED** [review]
It's good! But I think the main issue is that the paragraphs are too chunky and may be a bit tedious on the eyes - try split them up.

**Comment by IMB11** [comment]
ref: https://github.com/FabricMC/fabric-docs/pull/50#discussion_r1510295736

**Comment by IMB11** [comment]
I think this will be a very good introductory pull request adding some preliminary block stuff, I believe that data generation should be focused on [in a separate pull request, maybe going through the stages of data generation setup etc.] before going into complex detail about block models, states, loot tables etc.

**Review by IMB11 — DISMISSED** [review]

**Comment by IMB11** [comment]
Going to close and rewrite this.

**Pull request closed without merging**